src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -2416,15 +2416,27 @@ internal void ILGenForCreateInstance(CodeGenerator ilg, Type type, Type cast, bo
                 }
             }
 
-            MethodInfo Activator_CreateInstance = typeof(Activator).GetMethod(
-                  "CreateInstance",
-                  CodeGenerator.StaticBindingFlags,
-                  new Type[] { typeof(Type) }
-                  );
-            ilg.Ldc(type);
-            ilg.Call(Activator_CreateInstance);
-            if (cast != null)
-                ilg.ConvertValue(Activator_CreateInstance.ReturnType, cast);
+            Type formatterServiceType = typeof(string).GetTypeInfo().Assembly.GetType("System.Runtime.Serialization.FormatterServices");
+            MethodInfo getUninitializedObject = formatterServiceType != null
+                ? formatterServiceType.GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static)
```

Comment:
GetUninitializedObject does exactly what it says: it gets an object that hasn't been initialized, i.e. where no constructor has been run.  Is that really the right behavior?  Seems like it'd be very common to get an object in a really bad state, where functionality on the object relies on a constructor having initialized it appropriately.